java/core/src/java/org/apache/orc/impl/TreeReaderFactory.java

```diff
@@ -990,6 +1007,10 @@ public void nextVector(ColumnVector previousVector,
       TimestampColumnVector result = (TimestampColumnVector) previousVector;
       super.nextVector(previousVector, isNull, batchSize);
 
+      if (context.isUseUTCTimestamp()) {
+        result.setIsUTC(true);
```

Comment:
result.setIsUTC(context.isUseUTCTimestamp());

Just in case result is in UTC but context.isUseUTCTimestamp() is false.